docs/config.md

```diff
@@ -155,6 +155,17 @@ provided.  It must be an object containing keys:
     * *commonName* -- Required.  The common name to use for all TLS requests.
     * *caCertPath* -- Optional.  Use the given CA cert for common name
     validation.
+  * *io.l5d.clientTls.boundPath*: Determine the common name based on the
+  destination bound path.  This plugin supports the following options:
+    * *caCertPath* -- Optional.  Use the given CA cert for common name
+    validation.
+    * *names* -- Required.  A list of name matchers which each must contain:
+      * *prefix* -- A path prefix.  All destinations which match this prefix
+      will use this entry to determine the common name.  Wildcards and variable
+      capture are allowed (see: `io.buoyant.linkerd.util.PathMatcher`)
+      * *commonNamePattern* -- The common name to use for destinations matching
+      the above prefix.  Variables captured in the prefix may be used in this
+      string.
```

Comment:
At this point the bullet indentations are making documentation hard to read. (eg, here we are documenting commonNamePattern which is a required option for a list of names, which is a required option for a tls plugin, whose class name is specified as a required option for tls.) Let's make separate headered sections for each tls plugin